File: Com.Fubei.Api.OpenApi/Biz/FubeiOpenApiFactory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Castle.DynamicProxy;
using Com.Fubei.Api.OpenApi.Biz.AgentApi;
using Com.Fubei.Api.OpenApi.Biz.MerchantApi;
using Com.Illuminati.Galileo.Net.Interceptor;

namespace Com.Illuminati.Galileo.Biz
{
    public sealed class FubeiOpenApiFactory
    {
        static FubeiOpenApiFactory()
        {
            var proxyGenerator = new ProxyGenerator();
            MerchantApi = proxyGenerator.CreateInterfaceProxyWithoutTarget<IFubeiMerchantApi>(ProxyGenerationOptions.Default, new FubeiOpenApiInterceptor());
            AgentApi = proxyGenerator.CreateInterfaceProxyWithoutTarget<IFubeiAgentApi>(ProxyGenerationOptions.Default, new FubeiOpenApiInterceptor());
        }

        /// <summary>
        /// 开放平台1.0接口
        /// </summary>
        public static IFubeiMerchantApi MerchantApi { get; }


        /// <summary>
        /// 开放平台2.0接口
        /// </summary>
        public static IFubeiAgentApi AgentApi { get; }
    }
}

File: Com.Fubei.OpenApi.Vb/Biz/FubeiOpenApiFactory.vb
Imports Castle.DynamicProxy
Imports Com.Fubei.Api.OpenApi.Biz.V1
Imports Com.Fubei.Api.OpenApi.Biz.V2
Imports Com.Fubei.Api.OpenApi.Net.Interceptor

Namespace Biz

    ' 开放平台接口工厂类
    Public NotInheritable Class FubeiOpenApiFactory
        Shared Sub New()
            Dim proxyGenerator = New ProxyGenerator()
            MerchantApi = proxyGenerator.CreateInterfaceProxyWithoutTarget(Of IFubeiMerchantApi)(ProxyGenerationOptions.Default, New FubeiOpenApiInterceptor())
            AgentApi = proxyGenerator.CreateInterfaceProxyWithoutTarget(Of IFubeiAgentApi)(ProxyGenerationOptions.Default, New FubeiOpenApiInterceptor())
        End Sub

        ' 开放平台1.0接口
        Public Shared ReadOnly Property MerchantApi As IFubeiMerchantApi

        ' 开放平台2.0接口
        Public Shared ReadOnly Property AgentApi As IFubeiAgentApi
    End Class
End Namespace
